gossip/src/duplicate_shred_handler.rs

```diff
@@ -144,18 +165,61 @@ impl DuplicateShredHandler {
             Some(SlotStatus::Frozen) => {
                 return false;
             }
-            Some(SlotStatus::UnfinishedProof(slot_map)) => {
-                if let Some(proof_chunkmap) = slot_map.peek(&data.from) {
-                    if proof_chunkmap.wallclock < data.wallclock {
-                        return false;
+            Some(SlotStatus::UnfinishedProof(slot_map)) => match slot_map.get(&data.from) {
+                None => {
+                    if slot_map.len() < MAX_PUBKEY_PER_SLOT {
+                        return true;
                     }
+                    return self.check_has_enough_stake_and_cleanup(&slot, &data.from);
```

Comment:
could just pass `slot_map` instead of `slot`?

Reply:
I need a mutable reference now, so it's easier to just pass slot here.